spec/unit/provider/package/windows/registry_uninstall_entry_spec.rb

```diff
@@ -22,4 +24,40 @@
       expect(quiet_uninstall_string_key).to eql "QuietUninstallStringPath"
     end
   end
+
+  describe "when pass 'nil' to find_entries", :windows_only do
+    let(:package_name) { nil }
+    it "should return empty entries array" do
+      find_entries = Chef::Provider::Package::Windows::RegistryUninstallEntry.find_entries(package_name)
+      expect(find_entries.size).to eql 0
+    end
+  end
+
+  describe "when pass empty string to find_entries", :windows_only do
+    let(:package_name) { " " }
+
+    it "should return no entries" do
+      find_entries = Chef::Provider::Package::Windows::RegistryUninstallEntry.find_entries(package_name)
+      expect(find_entries.size).to eql 0
+    end
+  end
+
+  describe "when package not found", :windows_only do
+    let(:package_name) { "hive" }
+
+    it "should return no entries" do
+      find_entries = Chef::Provider::Package::Windows::RegistryUninstallEntry.find_entries(package_name)
+      expect(find_entries).to eql []
+    end
+  end
+
+  describe "When package found successfully", :windows_only do
+    let(:package_name) { "Chef Client" }
+
+    it "should return 'Chef Client' entries" do
+      allow(Chef::Provider::Package::Windows::RegistryUninstallEntry).to receive(:read_registry_property).and_return("Chef Client")
+      find_entries = Chef::Provider::Package::Windows::RegistryUninstallEntry.find_entries(package_name).first
+      expect(find_entries.display_name).to eql registry_uninstall_entry.first.display_name
```

Comment:
The purpose of referring to `registry_uninstall_entry.first.display_name` here isn't clear.